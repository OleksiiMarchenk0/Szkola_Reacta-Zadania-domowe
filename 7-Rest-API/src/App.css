*{
  box-sizing: border-box;
}
.App {
  text-align: center;
}

.User{
  padding: 30px;
  border-bottom: 1px solid #ccc;
}
.header{
  width: 100vw;
  color:blue;
  padding: 30px;
}
button{
  border: none;
  background: transparent;
}
a{
  text-decoration: none;
  color:blue;
}
h1{
  font-size: 20px;
}
.map{
  height: 30vh;
}