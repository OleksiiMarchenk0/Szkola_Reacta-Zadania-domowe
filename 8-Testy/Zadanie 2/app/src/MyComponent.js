import React, { useState, useRef } from "react";
MyComponent.defaultProps = {
  start: 0
}

function MyComponent({start}) {
 const [val,setVal]=useState(0);
  const textInput = useRef(null);
  const handleAddButton = () => {
    setCount(count + 1);
  };
  const Reset = () => {
    setCount(parseInt(start));
  };
  const handleSubstractButton = () => {
    setCount(count - 1);
  };
  const onSubmit = (e) => {
    e.preventDefault();
    let value = parseInt(textInput.current.value);
    setCount(value);
    textInput.current.value='';
  };
  const handleChange=(e)=>{
   setVal(e.target.value)
  }
  const [count, setCount] = useState(start);

  return (
    <div>
      <br />
      <form onSubmit={onSubmit}>
        <input id="num" onChange={handleChange} type="number" ref={textInput} value={val} placeholder="Wartośc" />
        <input className="submit" type="submit" value="Zmień" />
      </form>
      <button className="Reset" onClick={Reset}>Reset</button>
      <br />
      <br />
      <span className="counterVal">{`Count : ${count}`}</span>
      <button className="substract" onClick={handleSubstractButton}>-</button>
      <button className="add" onClick={handleAddButton}>+</button>
    </div>
  );
}

export default MyComponent;
