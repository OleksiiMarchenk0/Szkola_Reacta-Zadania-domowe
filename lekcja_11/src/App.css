.App {
  text-align: center;
}

.User{
  padding: 30px;
  border-bottom: 1px solid #ccc;
}