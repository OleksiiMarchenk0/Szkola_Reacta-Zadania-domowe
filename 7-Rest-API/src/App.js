import React from 'react';
import UserDetails from "./components/UserDetails/UserDetails.js";
import { BrowserRouter as Router, Switch, Route, Redirect } from "react-router-dom";
import "./App.css";
import UserList from "./components/UserList/UserList";

function App() {
  return (
    <div>
      <Router>
      <Redirect to="/users-list" />
        <Switch>
          <Route path="/users-list">
            <UserList />
          </Route>
          <Route path="/users/:id">
            <UserDetails />
          </Route>
        </Switch>
      </Router>
    </div>
  );
}

export default App;
