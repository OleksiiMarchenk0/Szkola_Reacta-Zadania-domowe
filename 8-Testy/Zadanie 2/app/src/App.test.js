/**-----------TESTOWANIE APLIKACJI-ZADANIE 2--------------- */
import App from "./App";
import { shallow } from "enzyme";
import MyComponent from "./MyComponent";
import { render } from "@testing-library/react";
describe("app test", () => {
  it("initial value should be 0", () => {
    const app = shallow(<App />);
    expect(app.find("MyComponent").prop("start")).toEqual(0);
  });
  it("initial value should be 10", () => {
    let component = shallow(<MyComponent start={10} />);
    let counterVal = component.find(".counterVal");
    expect(counterVal.text()).toBe("Count : 10");
  });
  it("should be buttons + and - ", () => {
    const wrapper = shallow(<MyComponent />);
    const substract = wrapper.find("div button.substract");
    expect(substract.text()).toBe("-");
    const add = wrapper.find("div button.add");
    expect(add.text()).toBe("+");
  });
  it("state should properly changes", () => {
    const wrapper = shallow(<MyComponent />);
    const substract = wrapper.find("div button.substract");
    let counterVal = wrapper.find("div span.counterVal");
    const add = wrapper.find(".add");
    add.simulate("click");
    counterVal = wrapper.find("div span.counterVal");
    expect(counterVal.text()).toBe("Count : 1");
    substract.simulate("click");
    counterVal = wrapper.find("div span.counterVal");
    expect(counterVal.text()).toBe("Count : -1");
  });
  it("state should properly changes after input value in input field ", () => {
   // nie dziala
    let wrapper = shallow(<MyComponent />)
  let input = wrapper.find('#num');
  wrapper.find('#num').prop('onChange', { target: { value: 3 } })
  expect(input.props().value).toEqual(3);
  let submit = wrapper.find('.submit');
  submit.simulate("click");
  let counterVal = wrapper.find("div span.counterVal");
  expect(counterVal.text()).toBe("Count : 3");
  });
  it("state should set 0 after reset ", () => {
    let wrapper = shallow(<MyComponent />);
    let counterVal = wrapper.find("span.counterVal");
    let reset = wrapper.find("button.Reset");
    let add = wrapper.find(".add");
    add.simulate("click");
    add.simulate("click");
    reset.simulate("click");
    counterVal = wrapper.find("span.counterVal");
    expect(counterVal.text()).toBe("Count : 0");
  });
});