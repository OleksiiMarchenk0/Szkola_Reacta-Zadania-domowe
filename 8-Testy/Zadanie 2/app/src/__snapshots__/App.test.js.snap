// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing application is initial state equals 0 1`] = `[Function]`;
